### PR TITLE
[OSDEV-894] Use ContriCleaner parser in create facility api endpoint

### PR DESCRIPTION
**Pull request by mazursasha1990**
[OSDEV-894](https://opensupplyhub.atlassian.net/browse/OSDEV-894): Use `contricleaner` parser into create facility API endpoint (`facilities_view_set.py`)

All the Contricleaner functionality is ready for implementation into the create facility API endpoint. After these changes, data uploaded by a user in JSON format will be parsed by Contricleaner and returned all necessary fields to create a new facility. In case the data provided by the user are not valid then return a list with error messages.

### CODE REVIEW

File: src/django/contricleaner/lib/serializers/row_serializers/row_clean_field_serializer.py [hunk 1]
```diff
@@ -8,7 +8,12 @@ def __init__(self, field: str, new_field: str) -> None:
         self.new_field = new_field
 
     def validate(self, row: dict, current: dict) -> dict:
-        clean_value = clean(row[self.field])
+        field = row.get(self.field)
+
+        if not field:
```
Comment (killev) [inline src/django/contricleaner/lib/serializers/row_serializers/row_clean_field_serializer.py:13]: This change seems to be wrong. If there's no value for clean data, it should return error.

File: src/django/contricleaner/lib/serializers/row_serializers/row_composite_serializer.py [hunk 2]
```diff
@@ -18,11 +18,11 @@
 class RowCompositeSerializer:
     def __init__(self, sector_cache: SectorCacheInterface):
         self.validators = [
+            RowRequiredFieldsSerializer(),
```
Comment (killev) [inline src/django/contricleaner/lib/serializers/row_serializers/row_composite_serializer.py:21]: I think this change is excessive.

File: src/django/contricleaner/lib/serializers/row_serializers/row_country_serializer.py [hunk 3]
```diff
@@ -4,8 +4,10 @@
 
 class RowCountrySerializer(RowSerializer):
     def validate(self, row: dict, current: dict) -> dict:
+        country = row.get("country", '')
```
Comment (killev) [inline src/django/contricleaner/lib/serializers/row_serializers/row_country_serializer.py:7]: it's better to move this line under the `try:` block

File: src/django/contricleaner/lib/serializers/row_serializers/row_clean_field_serializer.py [hunk 4]
```diff
@@ -8,7 +8,9 @@ def __init__(self, field: str, new_field: str) -> None:
         self.new_field = new_field
 
     def validate(self, row: dict, current: dict) -> dict:
-        clean_value = clean(row[self.field])
+        field = row.get(self.field, '')
```
[the remark this answers is not in the document]
Comment (mazursasha1990) [reply src/django/contricleaner/lib/serializers/row_serializers/row_clean_field_serializer.py:11]: It can't be reverted completely, because `clean` is supposed to receive a value. In the case when a user doesn't add any name or address, the `clean` function will return an error that does not follow the expectations. 
I propose this variant of the solution `clean_value = clean(row.get(self.field, ''))`

File: src/django/contricleaner/lib/serializers/row_serializers/row_composite_serializer.py (hunk 2 above)
Comment (mazursasha1990) [reply src/django/contricleaner/lib/serializers/row_serializers/row_composite_serializer.py]: Done

File: src/django/contricleaner/lib/serializers/row_serializers/row_country_serializer.py (hunk 3 above)
Comment (mazursasha1990) [reply src/django/contricleaner/lib/serializers/row_serializers/row_country_serializer.py]: Done

File: src/django/contricleaner/tests/test_source_parser_json.py
Comment (killev) [inline src/django/contricleaner/tests/test_source_parser_json.py:1]: I wonder why you need to add this tests as part of this pr?

File: src/django/countries/tests/test_country_code_parser.py
Comment (killev) [inline src/django/countries/tests/test_country_code_parser.py:1]: Please move this as separate PR

**Review by killev — CHANGES_REQUESTED** [review]
Please split changes into 2 PRs:

1. Changes in countries
2. Changes in ContriCleaner
3. Changes in api/facilities

File: src/django/countries/tests/test_country_code_parser.py
Comment (mazursasha1990) [reply src/django/countries/tests/test_country_code_parser.py:1]: Done

**Review by killev — APPROVED** [review]
WOW. Seems this can be merged!

**Review by roman-stolar — APPROVED** [review]
LGTM
